### DISCUSSION TITLE
How to Retrieve the Name of a Type Programmatically?

### DISCUSSION THREAD

**Discussion by bennycode** [microsoft/TypeChat#263] [Q&A]
The `createTypeScriptJsonValidator` function requires the name of a JSON target type from the schema. If I rename a type in the schema, I must update the corresponding JSON target type string. Failing to do so will result in the following error:

> Error: JSON validation failed: '"./schema"' has no exported member named 'XYZ'. 

Is there a way to programmatically obtain the name of a type without manually typing the string? Perhaps using reflection metadata or something similar?

**Comment by anisirji** [comment]
can you further share your code snippets.
